pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -16,119 +16,119 @@ package ossm
-		}
-		util.KubeDeleteContents("bookinfo", testenvDeploymentX86)
-		time.Sleep(time.Duration(30) * time.Second)
 	})
+}
 
-	t.Run("smcp_test_annotation_quote_injection", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Test SMCP annotation quote value injection")
-		if _, err := util.Shell(`kubectl -n %s patch smcp/%s --type=merge --patch="%s"`, meshNamespace, smcpName, InjectedAnnotationsSMCPPatch); err != nil {
-			t.Fatal(err)
-		}
-
-		if _, err := util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName); err != nil {
-			t.Fatal(err)
-		}
+func DeployTestSsl(t TestHelper, ns string) {
+	oc.ApplyString(t, ns, fmt.Sprintf(testSSLDeploymentWithAnnotation, env.GetTestSSLImage()))
+	oc.WaitDeploymentRolloutComplete(t, ns, "testenv")
+}
 
-		log.Log.Info("Check a pod annotations")
-		if env.Getenv("SAMPLEARCH", "x86") == "p" {
-			util.KubeApplyContents("bookinfo", testenvDeploymentP)
-		} else if env.Getenv("SAMPLEARCH", "x86") == "z" {
-			util.KubeApplyContents("bookinfo", testenvDeploymentZ)
-		} else {
-			util.KubeApplyContents("bookinfo", testenvDeploymentX86)
-		}
-		util.CheckPodRunning("bookinfo", "app=env")
-		podName, err := util.GetPodName("bookinfo", "app=env")
-		if err != nil {
-			t.Fatalf("Failed to get pod name: %v", err)
-		}
-		annotations, err := util.GetPodAnnotations("bookinfo", podName, 30)
-		if err != nil {
-			t.Fatalf("Failed to get annotations: %v", err)
-		}
-		log.Log.Infof("Checking annotation: %v", annotations["test1.annotation-from-smcp"])
-		if _, ok := annotations["test1.annotation-from-smcp"]; !ok {
-			t.Errorf("Failed to get annotations: test1.annotation-from-smcp: test1")
-		} else if annotations["test1.annotation-from-smcp"] != "test1" {
-			t.Errorf("Failed to get annotations: test1.annotation-from-smcp: test1")
-		}
-		log.Log.Infof("Checking annotation: %v", annotations["test2.annotation-from-smcp"])
-		if _, ok := annotations["test2.annotation-from-smcp"]; !ok {
-			t.Errorf("Failed to get annotations: test2.annotation-from-smcp: '[test2]'")
-		} else if annotations["test2.annotation-from-smcp"] != "[test2]" {
-			t.Errorf("Failed to get annotations: test2.annotation-from-smcp: '[test2]'")
-		}
-		log.Log.Infof("Checking annotation: %v", annotations["test3.annotation-from-smcp"])
-		if _, ok := annotations["test3.annotation-from-smcp"]; !ok {
-			t.Errorf("Failed to get annotations: test3.annotation-from-smcp: '{test3}'")
-		} else if annotations["test3.annotation-from-smcp"] != "{test3}" {
-			t.Errorf("Failed to get annotations: test3.annotation-from-smcp: '{test3}'")
+func GetPodAnnotations(t TestHelper, podLocator oc.PodLocatorFunc) map[string]string {
+	annotations := map[string]string{}
+	pod := podLocator(t)
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		output := shell.Executef(t, "kubectl get pod %s -n %s -o jsonpath='{.metadata.annotations}'", pod.Name, pod.Namespace)
+		error := json.Unmarshal([]byte(output), &annotations)
```

Comment:
`error`collides with the `error` built-in interface
```suggestion
		err := json.Unmarshal([]byte(output), &annotations)
```